ipatests/test_integration/test_trust.py

```diff
@@ -177,6 +160,73 @@ def test_upn_user_authentication_in_nonposix_trust(self):
         self.master.run_command(['kinit', '-C', '-E', self.upn_principal],
                                 stdin_text=self.upn_password)
 
+    def test_filter_users(self):
+        """filter_users option is applied if SSSD starts offline.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/3983
+        https://pagure.io/SSSD/sssd/issue/3978
+
+        When there are users in filter_users for nss provider then no
+        look up should be in data provider and NCE/USER/ipa_domain/user
+        should be added to negative cache.
+        """
+        tasks.kinit_admin(self.master)
+        aduser = 'testuser@{0}' .format(self.ad_domain)
+        fakeuser = 'some_user@some.domain'
+
+        exp_txt1 = 'Looking up [{0}] in data provider' .format(fakeuser)
+        exp_txt2 = 'Adding [NCE/USER/{0}/@{1}] to negative cache permanently'\
+            .format(self.master.domain.name, aduser)
+        exp_txt3 = 'NCE/USER/{0}/{1}] to negative cache permanently'\
+                   .format(self.ad_domain, aduser)
+
+        resolution = '{0}:{1}'.format(self.ad_domain, self.master.domain.name)
```

Comment:
variable name is misleading. It holds not resolution, but simply domains. So name like domains is better.